gr-uhd/include/gnuradio/uhd/rfnoc_graph.h

```diff
@@ -0,0 +1,135 @@
+    // \param skip_property_propagation Disable property propagation on this
+    //                                  connection (see the UHD documentation)
+    virtual void connect(const std::string& src_block,
+                         const std::string& dst_block,
+                         const bool skip_property_propagation = false) = 0;
+
+    //! Create an RX streamer
+    //
+    // Note: This streamer is not connected to anything after creation.
+    //
+    // See also the UHD documentation for uhd::rfnoc::rfnoc_graph::create_rx_streamer().
+    //
+    // \param num_ports Number of streaming ports
+    // \param args Stream args.
+    virtual ::uhd::rx_streamer::sptr
+    create_rx_streamer(const size_t num_ports, const ::uhd::stream_args_t& args) = 0;
+
+    //! Create a TX streamer
+    //
+    // Note: This streamer is not connected to anything after creation.
+    //
+    // See also the UHD documentation for uhd::rfnoc::rfnoc_graph::create_tx_streamer().
+    //
+    // \param num_ports Number of streaming ports
+    // \param args Stream args.
+    virtual ::uhd::tx_streamer::sptr
+    create_tx_streamer(const size_t num_ports, const ::uhd::stream_args_t& args) = 0;
+
+    //! Commit the graph and run initial checks
+    //
+    // See ::uhd::rfnoc::rfnoc_graph::commit() for more documentation.
+    virtual void commit() = 0;
+
+    /*! Return a valid block ID string, if it exists, or an empty string, if not
+     *
+     * This will check the available blocks on the connected devices and see if
+     * they match the block name, device ID, and block number.
+     */
+    virtual std::string get_block_id(const std::string& block_name,
+                                     const int device_select,
+                                     const int block_select) = 0;
+
+    //! Set time source on the specified motherboard
+    //
+    // Note: This is a convenience call, it directly dereferences the underlying
+    // motherboard controller.
+    //
+    // \param source Time source (e.g., "internal")
+    // \param mb_index Motherboard index, starting at 0
+    virtual void set_time_source(const std::string& source, const size_t mb_index) = 0;
+
+    //! Set clock source on the specified motherboard
+    //
+    // Note: This is a convenience call, it directly dereferences the underlying
+    // motherboard controller.
+    //
+    // \param source Clock source (e.g., "internal")
+    // \param mb_index Motherboard index, starting at 0
+    virtual void set_clock_source(const std::string& source, const size_t mb_index) = 0;
+
+    //! Return a reference to the block with a given block ID
+    //
+    // This allows to retrieve a shared pointer to a block controller with a
+    // limit. It keeps track of the references given out, and will throw an
+    // exception if called more than max_ref_count times. This is to help ensure
+    // that GNU Radio blocks don't share block controllers unintentionally.
```

Comment:
if that's the motivation, it'd be a sign of bad design: wouldn't you use a `unique_ptr` if you got that problem?